cmd/fs/helpers.go

```diff
@@ -46,6 +48,116 @@ func filerForPath(ctx context.Context, fullPath string) (filer.Filer, string, er
 	return f, path, err
 }
 
+const dbfsPrefix string = "dbfs:/"
+const volumesPefix string = "dbfs:/Volumes"
+
 func isDbfsPath(path string) bool {
-	return strings.HasPrefix(path, "dbfs:/")
+	return strings.HasPrefix(path, dbfsPrefix)
+}
+
+func getValidArgsFunction(
+	pathArgCount int,
+	onlyDirs bool,
+	filerForPathFunc func(ctx context.Context, fullPath string) (filer.Filer, string, error),
+	mustWorkspaceClientFunc func(cmd *cobra.Command, args []string) error,
+) func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		cmd.SetContext(root.SkipPrompt(cmd.Context()))
+
+		err := mustWorkspaceClientFunc(cmd, args)
+
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveError
+		}
+
+		filer, path, err := filerForPathFunc(cmd.Context(), toComplete)
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveError
+		}
+
+		isDbfsPath := isDbfsPath(toComplete)
+
+		wsc := root.WorkspaceClient(cmd.Context())
+		_, err = wsc.CurrentUser.Me(cmd.Context())
```

Comment:
Added this check to be consistent with this: https://github.com/databricks/cli/blob/a13d77f8eb29a6c7587509721217a137039f20d6/cmd/sync/completion.go#L49-L52

Happy to remove it, the calls will eventually fail.

Reply:
This is likely to make sure that credentials actually work and not fail with some error

Reply:
The latency can be important for the tabbing UX. There's no functional difference if we bail out because an error happened during `/scim/me` vs during a `/dbfs/list` call. Right now, in either case, we do not display the error to the user. 

IMO, we should change that, though, and show the error as an autocomplete suggestion. It would suck if the autocompletion silently does not work. It's a bit hacky, though, so I'd be fine with not doing this.

Reply:
Removed the current user test and relying on the ReadDir call to fail. Autocompletion is definitely quicker now.